### ISSUE TITLE
Nested notification policies display wrong wait information’s when using Override general timings

### ISSUE THREAD

**Issue by JDA88** [grafana/grafana#73573]
### What happened?

I tried to create various nested notification policies with Override general timings and the displayed values are wrong in the main windows

### What did you expect to happen?

Correct values in the main windows

### Did this work before?

No

### How do we reproduce it?

1. Notification policies > Create a new nested policy with Override general timings `Group wait` = 1m, `Group interval` = 2m, `Repeat interval` = 3m
2. Notification policies > Create a second nested policy under the first one with Override general timings `Group wait` = 10m, `Group interval` = 20m, `Repeat interval` = 30m
3. Notification policies > Create a second nested policy under the first one with Override general timings, `Repeat interval` = 999m

The result if the following:
![image](https://github.com/grafana/grafana/assets/67072186/7a7690bc-2a98-420e-a8b8-201d91ba5875)

Issues:
On all the policy the `Repeat interval` is missing from the description
On the third one even though only the `Repeat interval` have been overridden it display incorrect values for `Group wait` and `Group interval`

The good news is that even if the displayed values are wrong, the configuration files is ok, so the it’s working correctly

### Environment (with versions)?

Grafana: v10.2.0
OS: Grafana Cloud
Browser: Edge / Chrome

### Grafana platform?

I use Grafana Cloud

**Comment by tonypowa** [comment]
hi @JDA88 and thank you for creating this issue

> On all the policy the Repeat interval is missing from the description

Good point. We will double-check with the team to understand why  `repeat interval` was left out.

> On the third one even though only the Repeat interval have been overridden it display incorrect values for Group wait and Group interval

Nice catch.  It seems that ` Group wait` and `Group interval` are displaying the the values from the root policy (default policy). However the 2nd nested child policy is actually inheriting the values from the parent policy (the policy you created on your step 1). In fact, hoovering over the properties `inherited`, shows the actual/"correct" values being used.

From my end. I can also see this is true by looking at the json of the 2nd child item. The general timings override values are the ones inherited from the parent policy (1m, 2m, 3m):

``` json
 {
                    "object_matchers": [
                        [
                            "new_1_child_2",
                            "=",
                            "2"
                        ]
                    ],
                    "repeat_interval": "16h39m"
                }
            ],
            "group_wait": "1m",
            "group_interval": "2m",
            "repeat_interval": "3m"
        }...

```

**Comment by JDA88** [comment]
Hi @tonypowa, I'm glad you could reproduce the issue.

**Comment by gillesdemey** [comment]
I've created [a PR](https://github.com/grafana/grafana/pull/73675) to fix this, the inheritance algorithm was working as intended (as pointed out by @tonypowa – thanks! 🙏) but the UI was still applying defaults even if we've specified it to inherit in the summary description.

As for why the repeat interval was left out we've noticed that this timing option is less often inspected than the other ones and since we're trying to keep the amount of information to something that isn't overwhelming we've decided not to include it in the summary.
